relay-server/src/utils/rate_limits.rs

```diff
@@ -1043,30 +1005,49 @@ mod tests {
         }
     }
 
+    fn enforce_and_apply(
```

Comment:
Generalized all the code into this function, since I wanted tests to be leaner since a lot of repetition was there previously and some tests were also passing without calling the `apply_with_outcomes` so to avoid that risk in the future, I decided to have everything into one.